driver-sync/src/test/functional/com/mongodb/client/CrudProseTest.java

```diff
@@ -30,41 +35,83 @@
 import static java.lang.String.format;
 import static java.util.Arrays.asList;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 import static org.junit.Assume.assumeTrue;
 
-// See https://github.com/mongodb/specifications/tree/master/source/change-streams/tests/README.rst#prose-tests
-public class WriteConcernProseTest extends DatabaseTestCase {
+/**
+ * See https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.rst#prose-tests
+ */
+public class CrudProseTest extends DatabaseTestCase {
     private BsonDocument failPointDocument;
 
     @Before
     @Override
     public void setUp() {
-        assumeTrue(canRunTests());
         super.setUp();
     }
 
-    // Ensure that the WriteConcernError errInfo object is propagated.
+    /**
+     * 1. WriteConcernError.details exposes writeConcernError.errInfo
+     */
     @Test
     public void testWriteConcernErrInfoIsPropagated() {
+        assumeTrue(isDiscoverableReplicaSet() && serverVersionAtLeast(4, 0));
+
         try {
             setFailPoint();
             collection.insertOne(Document.parse("{ x: 1 }"));
         } catch (MongoWriteConcernException e) {
             assertEquals(e.getWriteConcernError().getCode(), 100);
-            assertTrue(e.getWriteConcernError().getCodeName().equals("UnsatisfiableWriteConcern"));
+            assertEquals("UnsatisfiableWriteConcern", e.getWriteConcernError().getCodeName());
             assertEquals(e.getWriteConcernError().getDetails(), new BsonDocument("writeConcern",
                     new BsonDocument("w", new BsonInt32(2))
                             .append("wtimeout", new BsonInt32(0))
                             .append("provenance", new BsonString("clientSupplied"))));
         } catch (Exception ex) {
-            fail(format("Incorrect exception thrown in test: %s" + ex.getClass()));
+            fail(format("Incorrect exception thrown in test: %s", ex.getClass()));
         } finally {
             disableFailPoint();
         }
     }
 
+    /**
+     * 2. WriteError.details exposes writeErrors[].errInfo
+     */
+    @Test
+    public void testWriteErrorDetailsIsPropagated() {
+        assumeTrue(serverVersionAtLeast(3, 2));
```

Comment:
JFYI: the spec [says _"Using a 5.0+ server"_](https://github.com/mongodb/specifications/commit/5141a8968a36cb4f29f203110c5d48aaaa281533#diff-0f732e30e0b0253a827e0fb457f5b27a1d80332db2e046d5a45587ba744e4455R257).

Reply:
Only one assertion (about details) is inappropriate for older servers, so just made that one conditional.